core/plugins/org.polarsys.capella.core.model.helpers/src/org/polarsys/capella/core/model/helpers/ComponentExt.java

```diff
@@ -1743,9 +1745,11 @@ public static ComponentArchitecture getRootComponentArchitecture(Component compo
    */
   public static List<Part> getSubParts(Component component, boolean useDeploymentLinks) {
     List<Part> result = new ArrayList<Part>();
-    if (useDeploymentLinks) {
-      for (Part part : component.getRepresentingParts()) {
-        result.addAll(PartExt.getDeployedParts(part));
+    if (component != null) {
+      if (useDeploymentLinks) {
+        for (Part part : component.getRepresentingParts()) {
+          result.addAll(PartExt.getDeployedParts(part));
+        }
       }
     }
     result.addAll(getSubParts(component));
```

Comment:
Can you move this line inside the `if` too? It will avoid creation of a new empty ArrayList and an addAll